File: Torneig_Mort_Spring/src/main/java/org/eljust/DTO/GrupDTO.java
package org.eljust.DTO;

import java.util.ArrayList;
import java.util.List;

import com.fasterxml.jackson.annotation.JsonProperty;

import lombok.Getter;
import lombok.Setter;

/**
 * Aquest GrupDTO es una classe la qual ens dona tota la informació rellevant de
 * la entitat Grup. La ferem servir al controlador de Grup concretament per a
 * obtindre la informació d'un grup.
 */
@Getter
@Setter
public class GrupDTO {
	private Long idGrup;
	private String nom;
	@JsonProperty("Equips")
	private List<EquipGrupDTO> elsEquipsGrups = new ArrayList<>();
	private List<PartitDTO> elsPartits = new ArrayList<>();

}

File: app_torneig_flutter/lib/models/grup_dto.dart
import 'package:app_torneig/models/equip_grup_dto.dart';
import 'package:app_torneig/models/partit_dto.dart';

class GrupDTO {
  int? id;
  String? nom;
  List<PartitDTO>? elsPartits;
  List<EquipGrupDTO>? elsEquips;

  GrupDTO({
    required this.id,
    this.nom,
    this.elsEquips,
    this.elsPartits,
  });

  GrupDTO.fromJSON(Map<String, dynamic> objecteJSON) {
    id = objecteJSON['idGrup'];
    nom = objecteJSON['nom'];
    elsEquips = objecteJSON['Equips'] != null
        ? List<EquipGrupDTO>.from(
            objecteJSON['Equips'].map((equip) => EquipGrupDTO.fromJSON(equip)))
        : null;
    elsPartits = objecteJSON['elsPartits'] != null
        ? List<PartitDTO>.from(objecteJSON['elsPartits']
            .map((partit) => PartitDTO.fromJSON(partit)))
        : null;
  }
}
